pandas/core/indexing.py

```diff
@@ -1880,15 +1928,26 @@ def _has_valid_type(self, key, axis):
                                               "is not available")
                 raise ValueError("iLocation based boolean indexing cannot use "
                                  "an indexable as a mask")
-            return True
+            return
 
         if isinstance(key, slice):
-            return True
+            return
         elif is_integer(key):
-            return self._is_valid_integer(key, axis)
+            self._validate_integer(key, axis)
+        elif isinstance(key, tuple):
+            # a tuple should already have been caught by this point
+            # so don't treat a tuple as a valid indexer
+            raise IndexingError('Too many indexers')
         elif is_list_like_indexer(key):
-            return self._is_valid_list_like(key, axis)
-        return False
+            # check that the key does not exceed the maximum size of the index
+            arr = np.array(key)
+            l = len(self.obj._get_axis(axis))
+
+            if len(arr) and (arr.max() >= l or arr.min() < -l):
+                raise IndexError("positional indexers are out-of-bounds")
+        else:
```

Comment:
can do w/o the else here

Reply:
No, I would have to add a

```python
else:
    return
```
branch to the ``if`` just above, because when the indexer is valid it does not return/raise